back/spec/acceptance/confirmations_spec.rb

```diff
@@ -4,52 +4,178 @@
+      code = user.email_confirmation_code
+      do_request(confirmation: { email: user.email, code: code })
+      assert_status 200
+      do_request(confirmation: { email: user.email, code: code })
+      assert_status 422
+    end
+  end
+
+  post 'web_api/v1/user/confirm_code_unauthenticated' do
+    with_options scope: :confirmation do
+      parameter :email, 'The email address of the user to confirm.'
+      parameter :code, 'The 4-digit confirmation code received by email.'
+    end
+
+    context 'when email does not exist' do
+      let(:user) { create(:user_with_confirmation) }
+      let(:email) { 'nonexistent@example.com' }
+
+      before do
+        RequestConfirmationCodeJob.perform_now user
+      end
+
+      example 'returns a not found status when the email does not exist' do
+        do_request(confirmation: { email: email, code: '1234' })
+        expect(status).to eq 422
+      end
+    end
+
+    context 'when email exists' do
+      let(:user) { create(:user_with_confirmation, password: nil) }
+
+      before do
+        RequestConfirmationCodeJob.perform_now user
+      end
+
+      include_examples 'confirmation code validation'
+
+      example 'does not allow confirming a user with password' do
+        user_with_password = create(:user_with_confirmation, password: 'password123')
+        code = user_with_password.email_confirmation_code
+        do_request(confirmation: { email: user_with_password.email, code: })
+        assert_status 422
+      end
+    end
+  end
+
+  post 'web_api/v1/user/confirm_code_authenticated' do
     with_options scope: :confirmation do
-      parameter :code, 'The 6-digit confirmation code received by SMS or email.'
+      parameter :code, 'The 4-digit confirmation code received by email.'
     end
 
-    context 'when not logged in' do
-      example 'returns a not authorized status passing a valid code' do
+    context 'when user is not authenticated' do
+      let(:user) { create(:user_with_confirmation, password: 'password123') }
+
+      example 'returns an unauthorized status when the user is not authenticated' do
         do_request(confirmation: { code: '1234' })
         expect(status).to eq 401
       end
     end
 
-    context 'when logged in' do
-      let(:user) { create(:user_with_confirmation) }
+    context 'when user is authenticated' do
+      let(:user) { create(:user_with_confirmation, password: 'password123') }
 
       before do
         header_token_for user
         RequestConfirmationCodeJob.perform_now user
       end
 
-      example 'returns an ok status passing the right code' do
-        do_request(confirmation: { code: user.email_confirmation_code })
+      include_examples 'confirmation code validation'
+
+      example 'allows confirming a user without password' do
+        user_without_password = create(:user_with_confirmation, password: nil)
+        header_token_for(user_without_password)
```

Comment:
We usually don't call `header_token_for` directly inside examples. The typical approach is to use:

```ruby
before { header_token_for(user) } # or something similar
```

Then define the memoized helpers at the appropriate level:

```ruby
let(:user) { create(...) }
```

Reply:
But in this case we only need it for this test, and it's inside of this shared examples group... so how should I set it up in the right way?